src/arrayMethodSort.js

```diff
@@ -4,8 +4,21 @@
  * Implement method Sort
  */
 function applyCustomSort() {
-  [].__proto__.sort2 = function(compareFunction) {
-    // write code here
+  [].__proto__.sort2 = function (
+    compareFunction = (a, b) => String(a) > String(b),
+  ) {
+    for (let i = 0; i < this.length - 1; i++) {
+      if (compareFunction(this[i], this[i + 1]) > 0) {
+        const currentElement = this[i];
+
+        this[i] = this[i + 1];
+        this[i + 1] = currentElement;
+
+        i = -1;
```

Comment:
Resetting `i` to `-1` after a swap is a valid approach to restart the loop, but it is inefficient. Consider using a flag to track whether any swaps were made and only restart the loop if necessary.